eta/core/video.py

```diff
@@ -1056,6 +1056,70 @@ def from_dict(cls, d):
             schema=schema)
 
 
+class VideoSetLabelsDirectory(DirectoryContainer):
+    '''VideoSetLabelsDirectory is a DirectoryContainer that represents a list
+    of VideoLabels on disk.
+
+    See eta.core.serial.DirectoryContainer for more information and please
+    USE WITH CARE.
+    '''
+
+    _ELE_CLS = VideoLabels
+
+    _ELE_CLS_FIELD = "LABELS_CLS"
+
+    _ELE_DIR_ATTR = "labels_dir"
+
+    _ELE_MAP_ATTR = "labels_map"
+
+
+class BigVideoSetLabels(VideoSetLabels):
+    '''BigVideoSetLabels uses a VideoSetLabelsDirectory to iterate over
+    VideoLabels on disk.
+
+    See eta.core.serial.DirectoryContainer for more information and please
+    USE WITH CARE.
+    '''
+
+    def __init__(self, videos=None, schema=None, labels_dir=None):
+        '''Either videos (a VideoSetLabelsDirectoy instance)
+        or labels_dir must be provided.
+
+        Args:
+            videos (VideoSetLabelsDirectory)
+            schema (VideoLabelsSchema
+            labels_str (str): directory path representing a
+                VideoSetLabelsDirectory
+        '''
+        if labels_dir is not None:
+            videos = VideoSetLabelsDirectory(labels_dir=labels_dir)
+        super(BigVideoSetLabels, self).__init__(videos=videos, schema=schema)
+
+    @classmethod
+    def from_dict(cls, d):
+        '''Constructs a  BigVideoSetLabels from a JSON dictionary.'''
+        videos = VideoSetLabelsDirectory.from_dict(d["videos"])
+
+        schema = d.get("schema", None)
+        if schema is not None:
+            schema = VideoLabelsSchema.from_dict(schema)
+
+        return cls(videos=videos, schema=schema)
+
+    def copy(self, new_videos_dir):
+        '''Copy the instance deeply.
+
+        Args:
+            new_images_dir (str): directory path to store the copy's data
+
+        Returns:
+            BigVideoSetLabels
+        '''
+        new_set = copy.deepcopy(self)
+        new_set.videos = self.videos.copy(new_videos_dir)
+        return new_set
+
```

Comment:
Will you have to override `VideoSetLabels.sort_by_filename()`?  Will `sorted()` do what you want it to do on a 'DirectoryContainer`?

Reply:
`sorted()` seems to work. I'll be doing more testing today, of course.

Reply:
I guess what I meant was that `sorted()` would return a `list` and not a `DirectoryContainer`.  Won't that cause problems for future operations?  (It violates the contract that `BigVideoSetLabels.videos` has to be a `VideoSetLabelsDirectory`.)

Reply:
Okay, yes. I can override `sort_by_filename` for the `Big*` classes.